### ISSUE TITLE
onClick Node

### ISSUE THREAD

**Issue by JMidd91** [retejs/rete#114]
Hi,

I am struggling on how to make a node using $().click() from jquery.

currently i am trying to do something similar to this - [https://codepen.io/Ni55aN/pen/MOYPEz](https://codepen.io/Ni55aN/pen/MOYPEz)

where I have a div elsewhere on screen, then I have a node that knows above that div it is then passed into the onClick node and then I want to have the output of that node have the event info when the div is clicked.

hope that makes sense!

**Comment by JMidd91** [comment]
hmm i still dont quite understand how tasks work at the moment, cant get my head around how the above codepen works

**Comment by Ni55aN** [comment]
Yesterday I have created task-plugin for v1.0.0 alpha. He provide more simple way for creating task-based editor. One of these days I will give an example of use.

The tasks are needed for deferred execution of instructions. At the time of processing, ordinary workers are running, which create a network of tasks that can be executed later at any time (for example, by an event)

**Comment by AaaCabbage** [comment]
@JMidd91 you can add click event when node was created. 
`
let newNode = node.builder(node.newNode())
newNode.el.onclick = function (ev) {
  //your operations
}
`

**Comment by JMidd91** [comment]
hello,

Think i am being miss understood here, i  dont want the click to be on my node,  but on the element one of my nodes is referring to. the button node has a drop down list of buttons that are on screen, you select one, then plug it into the on click node which you could then plug into what ever. your third node would then be triggered when you clicked your button.

**Comment by Ni55aN** [comment]
>  i dont want the click to be on my node

You can listen event on any element outside of editor.  The main thing you need to do is run an initial task

```js
yourEl.addEventListener('click', () => {
 task.run (someData);
});
```

but don't forget to clear previous listeners before the next process()

**Comment by AaaCabbage** [comment]
emm... my fault,
But i recommend you use [Vue.js](https://vuejs.org/index.html) instead of jquery, or you can use both. Vue give a much easier way to achieve the effect you want.

**Comment by JMidd91** [comment]
> but don't forget to clear previous listeners before the next process()

this seems to cause an issue when i want multiple tasks happening at once? the one created last is the only one alive

**Comment by Ni55aN** [comment]
@JMidd91 Tasks can not be executed at the same time, because all the same everything is done in one thread and the next chain of tasks can be executed only after the previous

If you want to run several chains of tasks, then you can create several instances of these Tasks

**Comment by JMidd91** [comment]
i think i am making several instances, im creating several nodes which create the tasks. its just the event listeners get deleted for me

**Comment by JMidd91** [comment]
ok so this was my way to fix things if anyone else has the same issue / has a way to improve it - 

`if($(inputs[0][0]).hasClass("hasEvent") != true){
                console.log("INPUT HAS NO EVENT ATTR");
                inputs[0][0].addEventListener('click', function (e) {
                    //console.log("click e", e)
                    //console.log("CLICK TASK", task)
                    //console.log("CLICK TASK OPTIONS", task.getOptions())
                    task.run(e);
                    task.reset();
                }, false);
                $(inputs[0][0]).addClass("hasEvent");
            }else{
                console.log("INPUT ALREADY HAS EVENT ATTACHED")
            }`

**Comment by JMidd91** [comment]
ok so this is the fix i come up with for reference if someone else has the same issue / a better fix for it - 

```javascript

            if($(inputs[0][0]).hasClass("hasEvent") != true){
                console.log("INPUT HAS NO EVENT ATTR");
                inputs[0][0].addEventListener('click', function (e) {
                    //console.log("click e", e)
                    //console.log("CLICK TASK", task)
                    //console.log("CLICK TASK OPTIONS", task.getOptions())
                    task.run(e);
                    task.reset();
                }, false);
                $(inputs[0][0]).addClass("hasEvent");
            }else{
                console.log("INPUT ALREADY HAS EVENT ATTACHED")
            }
```

**Comment by Ni55aN** [comment]
Good! The issue can be closed?

**Comment by JMidd91** [comment]
yup! cheers 👍
